### DISCUSSION TITLE
How does iced use tiny_skia on an Xrgb buffer?

### DISCUSSION THREAD

**Discussion by StratusFearMe21** [hecrj/iced#1915] [Q&A] [answered]
`tiny_skia` only supports drawing to an `RGBA` buffer, but iced uses it to draw to an `XRGB` buffer via `softbuffer`, how is the buffer being transformed between the 2 color spaces? I can't find the code in the codebase that would do that.

**Comment by hecrj** [comment, marked as answer]
`softbuffer` uses a `BGRX` buffer. We just swap the `r` and `b` color components when drawing with `tiny-skia`:

https://github.com/iced-rs/iced/blob/329fbc7b2157b84183849b2e0f600eb039435aed/tiny_skia/src/backend.rs#L494-L497

**Reply by StratusFearMe21** [reply]
Interesting, I would've thought this meant `XRGB`
> Pixel format (u32):
>
> 00000000RRRRRRRRGGGGGGGGBBBBBBBB
>
> 0: Bit is 0 R: Red channel G: Green channel B: Blue channel

Either way, your suggestion worked! Thanks for responding so quickly!
